mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -228,6 +228,29 @@ private[spark] object BLAS extends Serializable with Logging {
     }
     _nativeBLAS
   }
+ 
+  /**
+   * A := alpha * x * x^T + A
```

Comment:
Can you please add documentation here for each input value, including constraints about dimensions?

Also: Write ```x^T^``` for the Scala doc exponent (enclosing carats).